samples/minimum_criteria.rst

```diff
@@ -0,0 +1,98 @@
+.. _definition_and_minimum_criteria:
+
+Sample Definition and Minimum Criteria
+######################################
+
+Definition
+==========
+
+A sample is a basic Zephyr application that provides a brief overview of a feature, subsystem,
+or module. For example, Kernel services, drivers, protocols, etc.
+Samples should be limited in scope and should focus only on demonstrating non-trivial or
+essential aspects of the subsystem(s) or module(s) being highlighted.
+
+Minimum Sample Criteria
+=======================
+
+1. Samples must not be used as a testcase.
+++++++++++++++++++++++++++++++++++++++++++
+  * A feature should already have testcases under the tests category before the sample gets added.
+    Testcases are mandatory, and having samples may be optional.
+  * The primary purpose of a sample is to provide a reference to the user.
+  * Samples must not use Zephyr's testing framework.
+
+    * Must not use :kconfig:option:`CONFIG_ZTEST` or :kconfig:option:`CONFIG_ZTEST_NEW_API`.
+    * Must not use zasserts or asserts in samples.
+
+2. Samples should target multiple platforms and architectures.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize the use of ``platform_allow`` and architecture filters as it limits the scope
+    of testing to the mentioned platforms and architectures. Reference: `Testcase Docs`_
+  * Make use of ``integration_platforms`` to limit the scope when there are timing or
+    resource constraints.
+  * Make the sample as generic as possible. Avoid making a sample platform specific unless it is
+    for particular hardware.
+
+  .. _Testcase Docs: https://docs.zephyrproject.org/latest/develop/test/twister.html?highlight=test%20runner#test-cases.
+
+3. Twister should be able to build every sample.
+++++++++++++++++++++++++++++++++++++++++++++++++
+  * Every sample must have a YAML file.
+
+    **For example:**
+
+    .. code-block:: yaml
+
+      tests:
+        sample.kernel.cond_var:
+          integration_platforms:
+            - native_posix
+          tags: kernel condition_variables
+          harness: console
+          harness_config:
+            type: one_line
+            regex:
+              - ".*Waited and joined with 3 threads"
+
+  **Guidelines for Samples:**
+    * Minimize the use of ``platform_allow`` and architecture filters.
+    * Do not mark the test as build only unless necessary.
+    * Any test case-specific configuration options are added using ``extra_args`` or
+      ``extra_configs`` options in the YAML file. The prj.conf file should have the in-common
+      configuration options.
+    * There can be multiple testcases in the sample's YAML file to verify if the feature works
+      with different configurations. For example, different logging modes are tested using
+      multiple test scenarios in samples/subsys/logging/syst.
```

Comment:
As samples aren't test cases, perhaps the use of the word "tested" here should be avoided.

Reply:
I think this refers to testing the sample itself, i.e. test that the `hello world` does emit `hello world`, but it should not be used as to test an underlaying feature. The language should be changed here to reflect that.

Reply:
At least we need a entry in the test section of a sample yaml that "test builds" the same, if the sample that can provide output that can be verified, then output should be evaluated against and expected value to have some level of testing for the sample itself.

Reply:
@henrikbrixandersen @nashif Have updated this PR as per the requested changes. Please re-review. Thanks!